src/adaptix/_internal/morphing/generic_provider.py

```diff
@@ -150,17 +175,56 @@ def wrapped_loader_with_enums(data):
 
         return wrapped_loader_with_enums
 
+    def _get_literal_loader_with_bytes(
+        self,
+        basic_loader: Loader,
+        allowed_values: Collection,
+        bytes_loader: Loader,
+    ) -> Loader:
+        def wrapped_loader_with_bytes(data):
+            try:
+                bytes_value = bytes_loader(data)
+            except LoadError:
+                pass
+            else:
+                if bytes_value in allowed_values:
+                    return bytes_value
+            return basic_loader(data)
+
+        return wrapped_loader_with_bytes
+
+    def _get_literal_loader_many(self, *loaders: Loader, basic_loader: Loader) -> Loader:
+        if len(loaders) == 1:
+            return loaders[0]
+
+        def wrapped_loader_many(data):
+            for c, loader in enumerate(loaders):
+                try:
+                    return loader(data)
+                except LoadError:
+                    last_iteration = len(loaders) - 1
+                    if c != last_iteration:
+                        continue
+                    raise
```

Comment:
It looks like `raise` branch are not needed. `basic_loader` is last loader in chain and only it must raise `LoadError`